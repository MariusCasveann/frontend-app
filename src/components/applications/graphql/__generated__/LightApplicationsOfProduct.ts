/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: LightApplicationsOfProduct
// ====================================================

export interface LightApplicationsOfProduct_lightApplicationsOfProduct {
    __typename: 'LightApplicationDTO';
    id: any | null;
    name: string | null;
    service: boolean;
    staticPictureId: any | null;
}

export interface LightApplicationsOfProduct {
    lightApplicationsOfProduct: (LightApplicationsOfProduct_lightApplicationsOfProduct | null)[] | null;
}

export interface LightApplicationsOfProductVariables {
    productId?: any | null;
}
