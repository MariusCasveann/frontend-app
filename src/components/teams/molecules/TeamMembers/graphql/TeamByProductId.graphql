query TeamByProductId($productId: Long) {
    teamByProductId(productId: $productId) {
        id
        productOwner {
            firstName
            lastName
            id
        }
        teamMembers {
            firstName
            lastName
            id
            email
            roles {
                name
            }
        }
    }
}
