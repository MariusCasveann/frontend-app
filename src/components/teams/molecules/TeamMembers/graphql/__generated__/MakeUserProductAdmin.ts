/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: MakeUserProductAdmin
// ====================================================

export interface MakeUserProductAdmin {
    makeUserProductAdmin: boolean;
}

export interface MakeUserProductAdminVariables {
    productId?: any | null;
    userId?: any | null;
}
