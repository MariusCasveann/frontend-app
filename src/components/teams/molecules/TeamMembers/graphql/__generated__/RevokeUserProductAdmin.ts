/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: RevokeUserProductAdmin
// ====================================================

export interface RevokeUserProductAdmin {
    revokeUserProductAdmin: boolean;
}

export interface RevokeUserProductAdminVariables {
    productId?: any | null;
    userId?: any | null;
}
