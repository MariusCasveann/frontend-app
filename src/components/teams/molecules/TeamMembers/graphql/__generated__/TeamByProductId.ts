/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: TeamByProductId
// ====================================================

export interface TeamByProductId_teamByProductId_productOwner {
    __typename: 'User';
    firstName: string | null;
    lastName: string | null;
    id: any | null;
}

export interface TeamByProductId_teamByProductId_teamMembers_roles {
    __typename: 'Role';
    name: string | null;
}

export interface TeamByProductId_teamByProductId_teamMembers {
    __typename: 'User';
    firstName: string | null;
    lastName: string | null;
    id: any | null;
    email: string | null;
    roles: (TeamByProductId_teamByProductId_teamMembers_roles | null)[] | null;
}

export interface TeamByProductId_teamByProductId {
    __typename: 'Team';
    id: any | null;
    productOwner: TeamByProductId_teamByProductId_productOwner | null;
    teamMembers: (TeamByProductId_teamByProductId_teamMembers | null)[] | null;
}

export interface TeamByProductId {
    teamByProductId: TeamByProductId_teamByProductId | null;
}

export interface TeamByProductIdVariables {
    productId?: any | null;
}
