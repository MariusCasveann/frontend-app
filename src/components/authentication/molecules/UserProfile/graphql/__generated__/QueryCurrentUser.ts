/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { PermissionTypeEnum, SecuredEntityTypeEnum } from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: QueryCurrentUser
// ====================================================

export interface QueryCurrentUser_currentUser_roles_concretePermissions {
    __typename: 'EntityPermission';
    permissionType: PermissionTypeEnum | null;
    id: any | null;
    entityType: SecuredEntityTypeEnum | null;
    entityId: any | null;
}

export interface QueryCurrentUser_currentUser_roles {
    __typename: 'Role';
    id: any | null;
    name: string | null;
    readOnly: boolean | null;
    concretePermissions: (QueryCurrentUser_currentUser_roles_concretePermissions | null)[] | null;
}

export interface QueryCurrentUser_currentUser {
    __typename: 'UserDTO';
    activated: boolean;
    createdBy: string | null;
    createdDate: any | null;
    email: string | null;
    firstName: string | null;
    id: any | null;
    imageContent: string | null;
    langKey: string | null;
    lastModifiedBy: string | null;
    lastModifiedDate: any | null;
    lastName: string | null;
    login: string | null;
    roles: (QueryCurrentUser_currentUser_roles | null)[] | null;
    staticPictureId: any | null;
    subscribedToNewsletter: boolean | null;
}

export interface QueryCurrentUser {
    currentUser: QueryCurrentUser_currentUser | null;
}
