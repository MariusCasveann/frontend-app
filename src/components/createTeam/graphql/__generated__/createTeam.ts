/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { TeamInput } from './../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL mutation operation: CreateTeam
// ====================================================

export interface CreateTeam_createTeam {
    __typename: 'TeamDTO';
    id: any | null;
}

export interface CreateTeam {
    createTeam: CreateTeam_createTeam | null;
}

export interface CreateTeamVariables {
    productId?: any | null;
    team?: TeamInput | null;
}
