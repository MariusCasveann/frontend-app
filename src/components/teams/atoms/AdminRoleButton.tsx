import React from 'react';
import { Button } from 'antd';
import { TeamByProductId_teamByProductId_teamMembers_roles } from '../molecules/TeamMembers/graphql/__generated__/TeamByProductId';

interface AdminRoleButtonProps {
    isTeamMemberAdmin?: TeamByProductId_teamByProductId_teamMembers_roles | null;
    onUpdateRole: () => void;
}

export default ({ isTeamMemberAdmin, onUpdateRole }: AdminRoleButtonProps) => {
    const buttonLabel = (isTeamMemberAdmin && 'Revoke admin role') || 'Assign admin role';

    return (
        <Button type="link" size="small" onClick={onUpdateRole}>
            {buttonLabel}
        </Button>
    );
};
