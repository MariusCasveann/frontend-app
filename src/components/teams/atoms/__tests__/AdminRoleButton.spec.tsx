import * as React from 'react';
import renderer from 'react-test-renderer';
jest.mock('antd', () => ({ Button: 'Button' }));
import AdminRoleButton from './../AdminRoleButton';
import { TeamByProductId_teamByProductId_teamMembers_roles } from '../../molecules/TeamMembers/graphql/__generated__/TeamByProductId';

const mockMemberHasAdminRole: TeamByProductId_teamByProductId_teamMembers_roles = {
    name: 'a-admin',
    __typename: 'Role'
};

describe('AdminRoleButton', () => {
    describe('when team memeber is admin', () => {
        it('renders revoke right button', () => {
            const wrapper = renderer.create(<AdminRoleButton isTeamMemberAdmin={mockMemberHasAdminRole} />).toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });
    describe('when team memeber is not an admin', () => {
        it('renders assign right button', () => {
            const wrapper = renderer.create(<AdminRoleButton isTeamMemberAdmin={undefined} />).toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
