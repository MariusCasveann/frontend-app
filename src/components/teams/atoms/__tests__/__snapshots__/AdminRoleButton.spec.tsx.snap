// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminRoleButton when team memeber is admin renders revoke right button 1`] = `
<Button
  size="small"
  type="link"
>
  Revoke admin role
</Button>
`;

exports[`AdminRoleButton when team memeber is not an admin renders assign right button 1`] = `
<Button
  size="small"
  type="link"
>
  Assign admin role
</Button>
`;
